File: c++/ResponsibilityPattern/Majordomo.cpp
#include "stdafx.h"
#include "Majordomo.h"


Majordomo::Majordomo(string name) : Handler(name)
{
	this->name = name;
}


void Majordomo::HandeRequest(Request request)
{
	if (request.days > 0 && request.days < 3)
	{
		cout<< this->name << "批准了" << request.name + "的" << request.days << "天假期." << endl;
	}
	else
	{
		if (this->superior != NULL)
		{
			this->superior->HandeRequest(request);
		}
	}
}

File: csharp/DesignPattern.Responsibility/Majordomo.cs

using System;
namespace DesignPattern.Responsibility
{
    /// <summary>
    /// 总监
    /// </summary>
    public class Majordomo:Handler
    {
        public Majordomo(string name)
            : base(name)
        { 

        }

        public override void HandeRequest(Request request)
        {
            if (request.Days > 0 && request.Days < 3)
            {
                Console.WriteLine(this.Name + "批准了" + request.Name + "的" + request.Days + "天假期.");
            }
            else
            {
                if (this.superior != null)
                {
                    this.superior.HandeRequest(request);
                }
            }
        }
    }
}
